cmd/mtaBuild.go

```diff
@@ -295,6 +303,58 @@ func runMtaBuild(config mtaBuildOptions,
 	return err
 }
 
+func createMtaBuildSettingsInfo(config mtaBuildOptions, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment) error {
+	currentBuildSettingsInfo := BuildSettingsInfo{
+		GlobalSettingsFile: config.GlobalSettingsFile,
+		Profiles:           config.Profiles,
+		Publish:            config.Publish,
+		DefaultNpmRegistry: config.DefaultNpmRegistry,
+	}
+	var jsonMap map[string][]interface{}
+
+	if len(config.BuildSettingsInfo) > 0 {
```

Comment:
could the logic be "centralized"?
Then we could have one central test as well for the different scenarios like
* new file
* update file
* ...